script.js

```diff
@@ -0,0 +1,976 @@
+        this.language = localStorage.getItem('language');
+      } else {
+        this.language = 'engKeyName';
+      }
+    } catch (err) {
+      this.language = 'engKeyName';
+    }
+
+    const fragment = document.createDocumentFragment();
+
+    Object.keys(keyDirectory).forEach((key) => {
+      const keyElement = document.createElement('button');
+
+      keyElement.setAttribute('type', 'button');
+      keyElement.setAttribute('data-keycode', key.replace(/[a-z -]/g, ''));
+      keyElement.setAttribute('data-get-value', keyDirectory[key].getValue);
+      keyElement.classList.add('keyboard__key');
+
+      function createButtonStyle(nameButton, ...className) {
+        keyElement.classList.add(...className);
+        keyElement.textContent = nameButton;
+      }
+
+      switch (key) {
+        case 'keycode-8':
+          createButtonStyle('Backspace', 'key_large-wide');
+          break;
+        case 'keycode-9':
+          createButtonStyle('Tab', 'key_xs-wide');
+          break;
+        case 'keycode-20':
+          createButtonStyle('Caps Lock', 'key_small-wide', 'key_activated');
+          break;
+        case 'keycode-13':
+          createButtonStyle('Enter', 'key_xl-wide');
+          break;
+        case 'keycode-left-16':
+          createButtonStyle('Shift', 'key_large-wide');
+          break;
+        case 'keycode-right-16':
+          createButtonStyle('Shift', 'key_medium-wide');
+          break;
+        case 'keycode-32':
+          createButtonStyle('Space', 'key_xxl-wide');
+          break;
+        case 'keycode-left-17':
+        case 'keycode-right-17':
+          createButtonStyle('Ctrl');
+          break;
+        case 'keycode-left-18':
+        case 'keycode-right-18':
+          createButtonStyle('Alt');
+          break;
+        case 'keycode-91':
+          createButtonStyle('Win');
+          break;
+        case 'keycode-46':
+          createButtonStyle('Del');
+          break;
+        case 'language':
+          keyElement.setAttribute('data-keycode', key);
+          createButtonStyle(this.language.substring(0, 3).toUpperCase(), 'key_language');
+          break;
+        default:
+          keyElement.textContent = keyDirectory[key][this.language].standardKeyName;
+          break;
+      }
+      fragment.appendChild(keyElement);
+    });
+    return fragment;
+  }
+
+  keyPress(event, keyUp = false) {
+    const textArea = document.querySelector('.textarea');
+    const textPosition = textArea.selectionStart;
+    let keyPress = null;
+
+    if (event.type === 'mousedown' || event.type === 'mouseup') {
+      keyPress = event.target;
+    } else {
+      const keyRight = ['ShiftRight', 'AltRight', 'ControlRight'].includes(event.code);
+      const keysArray = keyRight ? Array.from(this.keys).reverse() : Array.from(this.keys);
+
+      keyPress = keysArray.find((key) => +key.dataset.keycode === event.keyCode);
+    }
+
+    function carriageMove(direction, number = 1) {
```

Comment:
You can also declare this function on level up as 'createKeys'. In this place you can just call this function.

Reply:
Roger that, too

Reply:
Done